### PR TITLE
Allow empty slots in mTextureCoords

### LINKED ISSUE #5632: Bug: Collada uvs from channels a,b,c are imported into Assimp channels 0,1,2.
**Describe the bug**
Collada: Now that we properly import uv channel numbers used by textures (merged today), it becomes apparent that second half of problem is importing data into correct uv channels. We properly find out that mesh contains data in channels a, b and c (example). But then d6aacefa1ef1219b85028b4038ef7e861c2e9462 changes a,b,c to 0,1,2 and we store data into wrong uv channels.

**To Reproduce**
1. Import and render .dae with texture mapped mesh that uses uv channel 1.
2. See that render is wrong, material wants data from channel 1 but data are stored in channel 0.

**Expected behavior**
Collada files should be imported with original uv channel numbers in both meshes and materials.

**Additional context**
I have problem locally fixed by reverting "empty slots are not allowed" paragraph of  d6aacefa1ef1219b85028b4038ef7e861c2e9462,
I'll send patch.
I'll also document that empty slots in mTextureCoords are ok... are they?
Also, I don't use GetNumUVChannels(), but I see it's implemented wrong. If we allow empty slots, it needs small fix.
Finally, I'm concerned about .dae files that contain channel numbers higher than AI_MAX_NUMBER_OF_TEXTURECOORDS. Do they exist? Are they common? (My approach in LightsprintSDK was to make array of channels resizable, so I never had to figure out how large numbers exist in wild.)

### PR DESCRIPTION
**Pull request by StepanHrbek**
This fixes #5632 by reverting undocumented assumption introduced in 2008.
It does not change API, it just explicitly states that empty slots are possible, after 2008 commits made them impossible without any trace of explanation why.
I understand if you don't want to change 16 years old status quo, there is risk of breaking userspace. But then the assumption about ordering of slots should be documented and fix of #5631 reverted.

### CODE REVIEW

**Comment by StepanHrbek** [comment]
Sorry, 3 older commits somehow slipped in, only 2 commits from today are relevant.

File: code/PostProcessing/ValidateDataStructure.cpp [hunk 1]
```diff
@@ -371,6 +371,7 @@ void ValidateDSProcess::Validate(const aiMesh *pMesh) {
         ReportWarning("There are unreferenced vertices");
     }
 
+#if 0 // no documentation found on why is it necessary
```
Comment (kimkulling) [inline code/PostProcessing/ValidateDataStructure.cpp:374]: Could you remove the code, please?

**Review by kimkulling — CHANGES_REQUESTED** [review]
Just one small finding.

File: code/PostProcessing/ValidateDataStructure.cpp (hunk 1 above)
Comment (StepanHrbek) [reply code/PostProcessing/ValidateDataStructure.cpp]: I removed 3 unwanted commits and that #if 0 block.
It's first time I'm using hard reset on public branch, is this how things should be done?

**Comment by StepanHrbek** [comment]
test/models/Collada/kwxport_test_vcolors.dae contains material with set="1" and mesh with input_set="1".
It means material wants uv channel 1 and mesh contains uv channel 1.

Before #5631, Assimp accidentally worked by importing both numbers as 0.
#5631 fix changes material number to 1.
#5632 fix changes also mesh number to 1.

It would be good to have test that checks that meshes have all uv channels requested by materials.
It would start failing on kwxport_test_vcolors.dae after #5631 fix, will stop failing after #5632 fix.

File: code/PostProcessing/ValidateDataStructure.cpp (hunk 1 above)
Comment (kimkulling) [reply code/PostProcessing/ValidateDataStructure.cpp]: You can just change it on your branch. We will merge this as one commit. So no hard reset will be necessary.

**Review by kimkulling — APPROVED** [review]

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**